Source/Core/Core/HW/SI/SI_DeviceAMBaseboard.cpp

```diff
@@ -0,0 +1,2361 @@
+                  data_out[data_offset++] = 0x31;  // 0x02
+                }
+                data_out[data_offset++] = 0x30;  // 0x03
+                break;
+              case CARDCommand::Clean:
+                data_out[data_offset++] = 0x02;  // 0x02
+                data_out[data_offset++] = 0x00;  // 0x03
+                break;
+              case CARDCommand::Load:
+                data_out[data_offset++] = 0x02;  // 0x02
+                data_out[data_offset++] = 0x30;  // 0x03
+                break;
+              case CARDCommand::SetShutter:
+                data_out[data_offset++] = 0x00;  // 0x02
+                data_out[data_offset++] = 0x00;  // 0x03
+                break;
+              }
+
+              data_out[data_offset++] = 0x30;  // 0x04
+              data_out[data_offset++] = 0x00;  // 0x05
+
+              data_out[data_offset++] = 0x03;  // 0x06
+
+              data_out[checksum_start] = data_offset - checksum_start;  // 0x00 len
+
+              u32 i;
+              data_out[data_offset] = 0;  // 0x07
+              for (i = 0; i < data_out[checksum_start]; ++i)
+                data_out[data_offset] ^= data_out[checksum_start + i];
+
+              data_offset++;
+
+              data_out[command_length_offset] = data_out[checksum_start] + 2;
+            }
+            else
+            {
+              for (u32 i = 0; i < length; ++i)
+                m_card_buffer[m_card_offset + i] = data_in[i];
+
+              m_card_offset += length;
+
+              // Check if we got a complete command
+
+              if (m_card_buffer[0] == 0x02)
+                if (m_card_buffer[1] == m_card_offset - 2)
+                {
+                  if (m_card_buffer[m_card_offset - 2] == 0x03)
+                  {
+                    m_card_command = m_card_buffer[2];
+
+                    switch (CARDCommand(m_card_command))
+                    {
+                    case CARDCommand::Init:
+                      NOTICE_LOG_FMT(SERIALINTERFACE_CARD, "GC-AM: Command CARD Init");
+
+                      m_card_write_length = 0;
+                      m_card_bit = 0;
+                      m_card_memory_size = 0;
+                      m_card_state_call_count = 0;
+                      break;
+                    case CARDCommand::GetState:
+                    {
+                      NOTICE_LOG_FMT(SERIALINTERFACE_CARD, "GC-AM: Command CARD GetState({:02X})",
+                                     m_card_bit);
+
+                      if (m_card_memory_size == 0)
+                      {
+                        std::string card_filename(File::GetUserPath(D_TRIUSER_IDX) + "tricard_" +
+                                                  SConfig::GetInstance().GetGameID().c_str() +
+                                                  ".bin");
+
+                        if (File::Exists(card_filename))
+                        {
+                          File::IOFile card = File::IOFile(card_filename, "rb+");
```

Comment:
```suggestion
                          File::IOFile card(card_filename, "rb+");
```
Unnecessarily verbose. There are two more cases of this in this file.